secboot/secboot_tpm_test.go

```diff
@@ -406,103 +406,120 @@ func (s *secbootSuite) TestUnlockIfEncrypted(c *C) {
 
 func (s *secbootSuite) TestSealKey(c *C) {
 	tmpDir := c.MkDir()
+	mockErr := errors.New("some error")
 
-	var mockEFI []string
-	for _, name := range []string{"a", "b", "c", "d", "e", "f"} {
-		mockFileName := filepath.Join(tmpDir, name)
-		err := ioutil.WriteFile(mockFileName, nil, 0644)
-		c.Assert(err, IsNil)
-		mockEFI = append(mockEFI, mockFileName)
-	}
+	for _, tc := range []struct {
+		tpmErr               error
+		tpmEnabled           bool
+		missingFile          bool
+		addEFISbPolicyErr    error
+		addSystemdEFIStubErr error
+		addSnapModelErr      error
+		provisioningErr      error
+		sealErr              error
+		provisioningCalls    int
+		sealCalls            int
+		expectedErr          string
+	}{
+		{tpmErr: mockErr, expectedErr: "cannot connect to TPM: some error"},
+		{tpmEnabled: false, expectedErr: "TPM device is not enabled"},
+		{tpmEnabled: true, missingFile: true, expectedErr: "file /does/not/exist does not exist"},
+		{tpmEnabled: true, addEFISbPolicyErr: mockErr, expectedErr: "cannot add EFI secure boot policy profile: some error"},
+		{tpmEnabled: true, addSystemdEFIStubErr: mockErr, expectedErr: "cannot add systemd EFI stub profile: some error"},
+		{tpmEnabled: true, addSnapModelErr: mockErr, expectedErr: "cannot add snap model profile: some error"},
+		{tpmEnabled: true, provisioningErr: mockErr, provisioningCalls: 1, expectedErr: "cannot provision TPM: some error"},
+		{tpmEnabled: true, sealErr: mockErr, provisioningCalls: 1, sealCalls: 1, expectedErr: "some error"},
+		{tpmEnabled: true, provisioningCalls: 1, sealCalls: 1, expectedErr: ""},
+	} {
+		var mockEFI []string
+		for _, name := range []string{"a", "b", "c", "d", "e", "f"} {
+			mockFileName := filepath.Join(tmpDir, name)
+			err := ioutil.WriteFile(mockFileName, nil, 0644)
+			c.Assert(err, IsNil)
+			mockEFI = append(mockEFI, mockFileName)
+		}
 
-	myParams := secboot.SealKeyParams{
-		ModelParams: []*secboot.SealKeyModelParams{
-			{
-				EFILoadChains:  [][]string{{mockEFI[0], mockEFI[1], mockEFI[2], mockEFI[3]}},
-				KernelCmdlines: []string{"cmdline1"},
-				Model:          &asserts.Model{},
-			},
-			{
-				EFILoadChains:  [][]string{{mockEFI[0], mockEFI[1], mockEFI[2]}, {mockEFI[3], mockEFI[4]}},
-				KernelCmdlines: []string{"cmdline2", "cmdline3"},
-				Model:          &asserts.Model{},
+		if tc.missingFile {
+			mockEFI[0] = "/does/not/exist"
```

Comment:
ah I see so you do need different files for each test case, in which case leave the loop in here, but I still think you want to have a per-test-case tmpdir

Reply:
The files are the same for all iterations, but for overall cleanliness and to avoid potential future problems caused by nonstandard practices I agree that it's better to have a per-test tmpdir.